### PR TITLE
Fixes slimes not being able to latch onto humans + faction tweak

### LINKED ISSUE #24412: Sentient slimes cannot attach to mobs
Likely related to the riding refactor.

### LINKED ISSUE #24746: Passive xenobio creatures are aggressive towards slimes
Maybe an oversight, maybe not, handle this issue however you deem necessary.

Nonetheless, the creatures spawned from injecting a gold slime core with blood start breaking stuff the moment they see a slime.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
fix: Slimes can now properly latch onto humans.
fix: Slimes won't aggro neutral mobs anymore. This includes blood-spawned gold slime mobs.
/:cl:

Fixes #24746.
Fixes #24412.

The faction won't affect slime aggro rules, i checked. Which is sorta bad, anyway.

Whoever made piggyback riding also made buckling mobs to humans SO MUCH MORE COMPLICATED I SPENT AN HOUR AND A HALF DEBUGGING THIS AND WHY DID YOU USE YES AS AN ARGUMENT NAME

### CODE REVIEW

**Comment by kevinz000** [comment]
Yes

**Comment by Kromgar** [comment]
FINALLY FUCKING FINALLY

**Comment by AnturK** [comment]
Change that name, also scream at oranges for merging it.

**Comment by XDTM** [comment]
What's wrong with the name?

**Comment by AnturK** [comment]
I mean the yes argument name.

**Comment by XDTM** [comment]
Removed the yes argument as a whole.

A round of applause for @kevinz000 for using isliving() to check if the target was not dead, and doing it twice on the same mob instead of on the riding human and on the ridden human.

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -898,14 +898,14 @@
 	.["Toggle Purrbation"] = "?_src_=vars;purrbation=\ref[src]"
 
 /mob/living/carbon/human/MouseDrop_T(mob/living/target, mob/living/user)
-	if((target != pulling) || (grab_state < GRAB_AGGRESSIVE) || (user != target) || !isliving(target) || !isliving(user))	//Get consent first :^)
+	if((target != pulling) || (grab_state < GRAB_AGGRESSIVE) || (user != target) || stat || user.stat)	//Get consent first :^)
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/human.dm]: these checks were necessary... atleast the target one was, it's any atom.

**Comment by PKPenguin321** [comment]
Hahaha nice going Kevin ya goon

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (XDTM) [inline code/modules/mob/living/carbon/human/human.dm]: should be ishuman then, unless we plan on allowing nonhumans to ride piggyback

Comment (XDTM) [inline code/modules/mob/living/carbon/human/human.dm]: or iscarbon for monkeys and xenos i guess

Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/human.dm]: save ian

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -898,7 +898,7 @@
 	.["Toggle Purrbation"] = "?_src_=vars;purrbation=\ref[src]"
 
 /mob/living/carbon/human/MouseDrop_T(mob/living/target, mob/living/user)
-	if((target != pulling) || (grab_state < GRAB_AGGRESSIVE) || (user != target) || stat || user.stat)	//Get consent first :^)
+	if((target != pulling) || (grab_state < GRAB_AGGRESSIVE) || (user != target) || !iscarbon(user) || stat || user.stat)//Get consent first :^)
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/human.dm]: I said save ian.

**Pull request merged**